gtsfm/common/gtsfm_data.py

```diff
@@ -387,3 +387,53 @@ def align_via_Sim3_to_poses(self, wTi_list_ref: List[Optional[Pose3]]) -> "Gtsfm
             aligned_data.add_track(track_aligned)
 
         return aligned_data
+
+    @classmethod
+    def align_data_to_axes(cls, gtsfm_data: "GtsfmData", walignedTw: Pose3) -> None:
+        """Align the GtsfmData's tracks and poses to the x,y,z axes.
+
+        Args:
```

Comment:
if we rewrite the function above as:
```python
    def align_via_Sim3_to_poses(self, wTi_list_ref: List[Optional[Pose3]]) -> "GtsfmData":
        """Align estimated, sparse multiview result (self) to a set of reference poses.
        Args:
            wTi_list_ref: list of reference/target camera poses, ordered by camera index.
        Returns:
            aligned_data: sparse multiview result that is aligned to the poses above.
        """
        # these are the estimated poses (source, to be aligned)
        wTi_list = self.get_camera_poses()
        # align the poses which are valid (i.e. are not None)
        # some camera indices may have been lost after pruning to largest connected component, leading to None values
        # rSe aligns the estimate `e` frame to the reference `r` frame
        wTi_list_aligned, rSe = geometry_comparisons.align_poses_sim3_ignore_missing(wTi_list_ref, wTi_list)
        return self.apply_Sim3(aSb=rSe)

    def apply_Sim3(self, aSb: Sim3) -> "GtsfmData" 
       aligned_data = GtsfmData(number_images=self.number_images())
        # update the camera pose to the aligned poses, but use the previous calibration
        for i, wTi in enumerate(wTi_list_aligned):
            if wTi is None:
                continue
            calibration = self.get_camera(i).calibration()
            aligned_data.add_camera(i, PinholeCameraCal3Bundler(wTi, calibration))
        # align estimated tracks to the ground truth
        for j in range(self.number_tracks()):
            # align each 3d point
            track_est = self.get_track(index=j)
            # place into the GT reference frame
            pt_ref = rSe.transformFrom(track_est.point3())
            track_aligned = SfmTrack(pt_ref)
            # copy over the 2d measurements directly into the new track
            for k in range(track_est.number_measurements()):
                i, uv = track_est.measurement(k)
                track_aligned.add_measurement(i, uv)
            aligned_data.add_track(track_aligned)

        return aligned_data
```
then this should be a 1 liner as follows:
```python
walignedTw = Similarity3(R=walignedTw.rotation(), t=walignedTw.translation(), s=1.0)
ba_output.apply_Sim3(walignedTw)
```
It looks like all of the code is pretty much the same.
```

Reply:
Thanks for writing out all this code for me :). I will update `align_via_Sim3_to_poses` in `gtsfm_data.py` to what you have and then add your new method called `apply_Sim3` to `gtsfm_data.py`